nautobot/ipam/models.py

```diff
@@ -238,6 +244,41 @@ def remove_virtual_machine(self, virtual_machine):
         instance = self.virtual_machines.through.objects.get(vrf=self, virtual_machine=virtual_machine)
         return instance.delete()
 
+    def add_virtual_device_context(self, virtual_device_context, rd="", name=""):
```

Comment:
This seems wrong to me - shouldn't `rd` and `name` default to `None`, not `""`? The latter will cause problems with the `(vdc, rd, name)` uniqueness constraint.

(edit) Actually it's worse than that, since `rd` is `null=True` but `name` is not. What gives? 🤢

Reply:
I guess let's leave this as-is for now since it matches the device and VM APIs already existing, but maybe open a followup item?

Reply:
https://github.com/nautobot/nautobot/issues/6945